### ISSUE TITLE
why prm loss use cross entropy not  reward?

### ISSUE THREAD

**Issue by yiyepiaoling0715** [OpenRLHF/OpenRLHF#522]
![image](https://github.com/user-attachments/assets/cdde2bd5-851f-4d99-8831-810c4a8e9fca)
prm is progress reward model,is a rl method, so the loss for optimizer  should be a reward value, but here is a cross entropy to coverage  logits of  positive and negative label mapped. so ti becomes a multilabel classification.
why progress reward model is as this ?

**Comment by hijkzzz** [comment]
better accuracy

**Comment by garyzhang99** [comment]
> Generally speaking, GenRM performs better, but contributions of a linear head version of PRM are welcome.

I wander if there are more results or insights supporting the claim that "GenRM performs better"? Because in the original GenRM paper, it seems to mention that GenRM without COT is only "slightly better or comparable" on GSM8K. 
<img width="674" alt="image" src="https://github.com/user-attachments/assets/ea5c818b-aba2-4793-8505-8f17214aff82" />


Also, it seems a bit counter-intuitive that using token probabilities performs better than using linear head.
